sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -230,8 +238,23 @@ class HashJoinIterator(
         // We are about to return something. We got everything we need from it so now let it spill
         // if there is more to be gathered later on.
         gathererStore.foreach(_.allowSpilling())
+        joinOutputRows += ret.get.numRows()
+      }
+
+      if (condition.isDefined && ret.isDefined) {
```

Comment:
Nit: I think this can be simplified since `ret.flatMap` will return `None` if ret is not defined.
```suggestion
      if (condition.isDefined) {
```